### PR TITLE
fix(attributes-to-props): don't convert value of option tag

### LINKED ISSUE #625: `value=` shouldn't be converted to `defaultValue=` for `<option>`
The value of `<option>` tags of `<select>` are fixed. Converting their `value=` to `defaultValue=` results bug if the option label is different from the value. For example:

```html
<select name="range" value="gte-2">
  <option value="lt-1">Less than 1</option>
  <option value="gte-2">≥2<option>
</select>
```

The value of `<select>`'s "gte-2" won't get a match and it will fallback to "Less than 1".

https://github.com/remarkablemark/html-react-parser/blob/cffffc71fe8e72e6a662de50c0b00a3d77c60cc6/lib/attributes-to-props.js#L42-L49

### PR DESCRIPTION
**Pull request by adammockor**
## What is the motivation for this pull request?

Fixes #625 and closes #790. Not sure if there is another element affected with this issue then `option`.

## What is the current behavior?

`attributesToProps` converts `value` attribute to `defaultValue` of `option` element what is not desirable.

## What is the new behavior?

`value` attribute is preserved on `option` element.

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [ ] Documentation
- [x] Types

### CODE REVIEW

File: test/attributes-to-props.test.js [hunk 1]
```diff
@@ -186,6 +186,12 @@ describe('attributesToProps with HTML attribute', () => {
       expect(attributesToProps(attributes)).toEqual(props);
     }
   );
+
+  it('preserve value of option element', () => {
```
Comment (remarkablemark) [inline test/attributes-to-props.test.js:190]: Can you pluralize the verb?

```suggestion
  it('preserves value of option element', () => {
```

File: lib/attributes-to-props.d.ts [hunk 2]
```diff
@@ -12,4 +12,7 @@ export type Props = Record<string, string> & {
  * @param attributes - HTML/SVG DOM attributes.
  * @returns - React props.
  */
-export default function attributesToProps(attributes: Attributes): Props;
+export default function attributesToProps(
```
Comment (remarkablemark) [inline lib/attributes-to-props.d.ts:15]: Can you fix the TypeScript tests?

```sh
npm run lint:dts
```

File: yarn.lock [hunk 3]
```diff
@@ -0,0 +1,5888 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (remarkablemark) [inline yarn.lock:1]: Can you remove `yarn.lock`?

**Review by remarkablemark — CHANGES_REQUESTED** [review]
Looks good overall, left some comments

**Comment by remarkablemark** [comment]
@adammockor You'll also need to update `.size-limit.json` since it has increased:

```sh
npm run size-limit
```

**Comment by adammockor** [comment]
Thank you for review.

About the API, I am not sure how to approach it. I made second parameter `nodeName` optional - so it's no breaking change, but this way it's not obvious why nodeName should be passed. Could be better to pass `node` into the function (and rename it to `nodeAttributesToProps`). You could consider it for next major release.

File: lib/attributes-to-props.d.ts [hunk 4]
```diff
@@ -12,4 +12,4 @@ export type Props = Record<string, string> & {
  * @param attributes - HTML/SVG DOM attributes.
```
Comment (remarkablemark) [inline lib/attributes-to-props.d.ts:12]: Could you add:

```
 * @param nodeName - DOM node name.
```

File: lib/attributes-to-props.js [hunk 5]
```diff
@@ -5,9 +5,10 @@ var utilities = require('./utilities');
  * Converts HTML/SVG DOM attributes to React props.
  *
  * @param {object} [attributes={}] - HTML/SVG DOM attributes.
+ * @param {string} [nodeName] - DOM node name. Help's to have smarter behavior while converting DOM attributes to props.
```
Comment (remarkablemark) [inline lib/attributes-to-props.js:8]: Could you fix grammar `Helps` or remove the 2nd sentence since `DOM node name.` seems self-explanatory.

**Review by remarkablemark — CHANGES_REQUESTED** [review]
Thanks for addressing my previous comments. I've left 2 more comments.

**Comment by remarkablemark** [comment]
> Thank you for review.
> 
> About the API, I am not sure how to approach it. I made second parameter `nodeName` optional - so it's no breaking change, but this way it's not obvious why nodeName should be passed. Could be better to pass just `node` into the function (and rename it to `nodeAttributesToProps`). You could consider it for next major release.

I think the 2nd parameter is fine since it's optional and backwards compatible. Thanks for opening this PR!

**Review by remarkablemark — APPROVED** [review]
LGTM

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/791#issuecomment-1372888548

**Pull request merged**